src/views/CaseDetail.vue

```diff
@@ -0,0 +1,246 @@
+<template>
+  <div class="case-detail z-index-3">
+    <div
+      class="container"
+      v-if="details"
+    >
+      <h1 class="passcode">
+        Case: <input ref="passcode" type="text" readonly :value="details.passcode"  @click="copyToClipboard"> 
```

Comment:
If the value of the `passcode` is not meant to be modified, why use  an `input`? Wouldn't be better if we use a `span` or `div`?. We could make the look & feel similar to the one of an input.

What do you think?

Reply:
Physical input is always more clickable and selectable than other elements on the screen. But I'd like to ask, why we have `readonly` attribute on the inputs, then?

On the other hand, `copyToClibboard()` function works with this input to copy the code to the clipboard automatically. If we pass into a div element, [we'll be creating a virtual `input` to put on DOM on every copy+paste action.](https://gist.github.com/Chalarangelo/4ff1e8c0ec03d9294628efbae49216db#file-copytoclipboard-js-L1), which I do not prefer [as we already can do this action without creating an element on the dom.](https://github.com/anonymouslyreportingtool/webapp/blob/feature/company_select_option_on_reportpage/src/views/CaseDetail.vue#L175)

Reply:
In general, I would prefer to avoid using `readonly` inputs if we do not expect any time on which you will write or edit them. I think that it is a great attribute to have if you have a conditional field. For instance, you could have a field that should only receive data depending on a checkbox field being true.

But if the `copyToClipboard` works better with an input field, that's great to me.

Reply:
> In general, I would prefer to avoid using readonly inputs if we do not expect any time on which you will write or edit them. I think that it is a great attribute to have if you have a conditional field. For instance, you could have a field that should only receive data depending on a checkbox field being true.

I do agree with this idea. 

> But if the copyToClipboard works better with an input field, that's great to me.
For now, I will stay as it is - but I will work on it again later for sure.